src/main/java/com/redhat/devtools/intellij/common/utils/ConfigHelper.java

```diff
@@ -109,12 +136,122 @@ public static boolean areEqual(NamedContext thisContext, NamedContext thatContex
         } else if (thatContext == null) {
             return false;
         }
+        if (!Objects.equals(thisContext.getName(), thatContext.getName())) {
+            return false;
+        }
 
-        return Objects.equals(thisContext.getContext(), thatContext.getContext());
+        return areEqual(thisContext.getContext(), thatContext.getContext());
+    }
+
+    private static boolean areEqual(Context thisContext, Context thatContext) {
+        if (thisContext == null) {
```

Comment:
So it the token has changed in the kubeconfig that will still return true ?

Reply:
@jeffmaury yes, because the context doesn't hold the token. The context only points to the user that holds the token. Only the kubeconfig holds the users. The client config does not store those, it thus holds the token the token in a separate property `oauthToken`.